exporter/datadogexporter/logs_exporter.go

```diff
@@ -114,3 +121,39 @@ func (exp *logsExporter) consumeLogs(ctx context.Context, ld plog.Logs) (err err
 	payloads := exp.translator.MapLogs(ctx, ld)
 	return exp.sender.SubmitLogs(exp.ctx, payloads)
 }
+
+// newLogsAgentExporter creates new instances of the logs agent and the logs agent exporter
+func newLogsAgentExporter(
+	ctx context.Context,
+	params exporter.CreateSettings,
+	cfg *Config,
+	sourceProvider source.Provider,
+) (logsagentpipeline.LogsAgent, exporter.Logs, error) {
+	logComponent := newLogComponent(params.TelemetrySettings)
+	cfgComponent := newConfigComponent(params.TelemetrySettings, cfg)
+	logsAgentConfig := &logsagentexporter.Config{
+		OtelSource:    otelSource,
+		LogSourceName: logSourceName,
+	}
+	hostnameComponent, err := logs.NewHostnameService(ctx, sourceProvider)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize logs agent hostname service: %w", err)
+	}
+	logsAgent := logsagentpipelineimpl.NewLogsAgent(logsagentpipelineimpl.Dependencies{
+		Log:      logComponent,
+		Config:   cfgComponent,
+		Hostname: hostnameComponent,
+	})
+	err = logsAgent.Start(ctx)
```

Comment:
We are starting the logsAgent during component creation time. We should not do this, that's what [`exporterhelper.WithStart`](https://pkg.go.dev/go.opentelemetry.io/collector/exporter/exporterhelper#WithStart) is used for

Reply:
Hm, so this has come up recently on open-telemetry/opentelemetry-collector/issues/10031. It's true that there are other things doing the wrong thing in our exporter, I will file a ticket for those. If we can solve this problem now for this one though it would be great not to make the problem worse.